### DISCUSSION TITLE
[Ideas] [Feature Request] Hide or disable "File Path" in "Get Info" box for any specific song

### DISCUSSION THREAD

**Discussion by leeee-me** [navidrome/navidrome#2913] [Ideas]
For some privacy or security consideration, I suggested we could have something of environment variable like:
ND_SHOWFILEPATHINGETINFO=true
and it can be turned off via value false, and then it would un-display the "file path" In get-info box.

Yes, although the file path can be seen and it cannot be used via curl or wget as URL even with correct or existed login authentication, yet we shall not reveal (or expose) the file structures of song arrangement on the file system.

**Comment by metalheim** [comment]
File path *should* return a fake path anyways (derived from tracks: /music/albumartist/album/title.extension) unless you specifially enable "Report Real path" for a particular client.
Not sure if maybe as admin Account you see the Real path every where. 

Some form of path has to be exposed to not break compatability with the subsonic API, so I think its a good compromise.

**Reply by leeee-me** [reply]
Hi @metalheim 
I tried both normal user and admin user, and they all did not turn on the "report real path" on any specific browser configuration.

![image](https://github.com/navidrome/navidrome/assets/41778946/48c81dff-2168-48bc-b428-b1030e5667dc)

But I am running Navidrome by docker on a Raspberry Pi 3, and here is my docker-run:

```
docker run -d --name navidrome --restart=unless-stopped \
--user $(id -u):$(id -g) \
-v $HOME/.music:/music \
-v /opt/navidrome/data:/data \
-p 127.0.0.1:4533:4533 \
-e TZ=CST-8 \
-e ND_LOGLEVEL=debug \
-e ND_ENABLEDOWNLOADS=false \
-e ND_ENABLEFAVOURITES=false \
-e ND_ENABLESTARRATING=false \
-e ND_LASTFM_ENABLED=false \
-e ND_LISTENBRAINZ_ENABLED=false \
-e ND_PASSWORDENCRYPTIONKEY=`gpg --list-public-keys $USER | head -n 2 | tail -n 1 | sed "s/^[ ]\+//g"` \
-e ND_UIWELCOMEMESSAGE="Heavens Burns Red" \
-e ND_IMAGECACHESIZE=0 \
-e ND_COVERARTPRIORITY='"cover.*, folder.*, embedded"' \
deluan/navidrome:latest
```

And I placed those MP3 files at $HOME/.music with some specific directory structures:

```
~$ ls -1 $HOME/.music/HBRv1/D1
'D1_(01)_基地の日常_-morning-_-_MANYO.mp3'
'D1_(02)_基地の日常_-noon-_-_MANYO.mp3'
'D1_(03)_基地の日常_-night-_-_MANYO.mp3'
'D1_(04)_31A_is_Here_Ⅰ_-_麻枝准.mp3'
'D1_(05)_暴走列車_-_MANYO.mp3'
'D1_(06)_天才少女たちのルーティン_-_MANYO.mp3'
'D1_(07)_通常運転_-_MANYO.mp3'
'D1_(08)_押し花の栞_-_MANYO.mp3'
'D1_(09)_濃霧注意報_-_MANYO.mp3'
'D1_(10)_Let'\''s_Shop!_-_MANYO.mp3'
'D1_(11)_剣はペンより強し_-_MANYO.mp3'
'D1_(12)_未踏の地_-_MANYO.mp3'
'D1_(13)_敵影_-_MANYO.mp3'
'D1_(14)_臨戦態勢_-_MANYO.mp3'
'D1_(15)_勝利の美酒_-_MANYO.mp3'
'D1_(16)_勝利の雄叫び_-_MANYO.mp3'
'D1_(17)_告解_-_MANYO.mp3'
'D1_(18)_銀時計_-_麻枝准.mp3'
'D1_(19)_After_You_Sleep_-_rionos.mp3'
folder.jpg

```

And I checked the GET INFO for any one of the exposed music:

![image](https://github.com/navidrome/navidrome/assets/41778946/2a689309-5291-4afd-8734-0100156050fc)

As you could see that information shows the same directory structure and real MP3 file name arranged on the host RPi3's file system.

Any idea why this happened? (only issue caused by docker version?) I installed Navidrome by Lighttpd's reverse proxy to leverage SSL connection for security.

**Reply by metalheim** [reply]
Mhm that shouldn't happen. 
As i understand it, the "report real path" option (default: unchecked) is exactly meant to obfuscate the real path for privacy reasons.

@deluan any idea why the fakePath isnt returned in above example, but the mf.Path?

**Reply by deluan** [reply]
> @deluan any idea why the fakePath isnt returned in above example, but the mf.Path?

The "Report Real Path" only works for Subsonic clients, not for Navidrome's UI. There is a isse open to have this option (hide paths from the Native API / UI), but it won't be worked until we can finish the new scanner and the new UI, which both will handle file paths differently

**Reply by leeee-me** [reply]
In that case, then I suggest to have a quick workaround to fix it temporarily as I mentioned earlier, to "hide" this "File path:" in UI via specific "flag" when generating the UI's HTML codes (RESTful API? Sorry I do not study it the API) at the end of engine, such as 
`ND_SHOWFILEPATHINGETINFO={true|false}`
And this will not impact the native API or any existed folder scanning mechanism until the root cause is solved accordingly. How about this idea?
